src/Akeneo/Pim/Enrichment/Product/back/API/Command/UserIntent/SetTableValue.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Akeneo\Pim\Enrichment\Product\API\Command\UserIntent;
+
+use Webmozart\Assert\Assert;
+
+/**
+ * @copyright 2022 Akeneo SAS (https://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+final class SetTableValue implements ValueUserIntent
+{
+    /**
+     * @param array<int, array<string, mixed>> $tableValue
+     *     ex. [
+     *             0 => ['ingredient' => 'butter', 'quantity' => 2],
+     *             1 => ['ingredient' => 'salt'],
+     *         ]
+     */
+    public function __construct(
+        private string $attributeCode,
+        private ?string $channelCode,
+        private ?string $localeCode,
+        private array $tableValue
+    ) {
+        Assert::allIsArray($tableValue);
```

Comment:
```suggestion
        Assert::isList($tableValue);
        Assert::allIsMap($tableValue);
```
(this way you can validate that the keys of the rows are integers, and the keys of the cells are strings, so the phpDoc is always correct